src/components/shared/Entity/EndpointConfig/index.tsx

```diff
@@ -99,37 +98,64 @@ function EndpointConfig({
 
     // Storing flag to handle knowing if a config changed
     //  during both create or edit.
-    const resetEndpointConfig = useMemo(
-        () =>
-            // TODO (connectors) We need to break this single flag into at least two
-            //  1 - if we want to reset data (only should happen on create when connector changes)
-            //  2 - if we need to update the schema (can happen on create and in edit)
-            editWorkflow && !unsupportedConnectorVersion
-                ? false
-                : connectorTag?.endpoint_spec_schema &&
-                  !isEqual(connectorTag.endpoint_spec_schema, endpointSchema),
-        [
-            connectorTag?.endpoint_spec_schema,
-            editWorkflow,
-            endpointSchema,
-            unsupportedConnectorVersion,
-        ]
-    );
+    const [resetEndpointConfig, updateSchema] = useMemo(() => {
+        let resetConfigResponse = false;
+        let updateSchemaResponse = false;
+
+        const schemaChanged = Boolean(
+            connectorTag?.endpoint_spec_schema &&
+                !isEqual(connectorTag.endpoint_spec_schema, endpointSchema)
+        );
+
+        if (editWorkflow) {
+            // In edit we never want to clear the config since any changes to
+            //  the schema should come from an upgrade and connectors should
```

Comment:
> any changes to the schema should come from an upgrade

The use of _upgrade_ is a little ambiguous here; I would specify that the _thing_ being upgraded is the connector version. That said, I don't understand why a comment related to the endpoint schema is made here instead of deferring to comment block above the `updateSchemaResponse` assignment. Noting that connectors will be backwards compatible as of Q1 2024 fits here.